tools/validate_spec.py

```diff
@@ -79,81 +77,106 @@ def cli(argv: List[str]) -> Union[int, str]:
-    path_invalid: Optional[Path] = args.directory_invalid
-    path_valid: Optional[Path] = args.directory_valid
-    full_output_path: Optional[Path] = args.json_output
-    abort_on_error: bool = args.abort_on_error
-    no_verification: bool = args.no_verification
-
-    path_generators: Dict[Path, Iterator[Path]] = {}
-    for path in [path_valid, path_invalid]:
-        if path is not None:
-            path_generators[path] = path.glob("*")
+        return message_path.read_bytes()
 
-    try:
-        identifier = ID(root_message_id)
-    except RecordFluxError as e:
-        raise ValidationError(f'invalid identifier "{root_message_id}" : {e}') from e
+    def rflx_parse_message_from_bytes(message: bytes) -> Union[str, MessageValue]:
+        pdu_model: MessageValue = pdu_message.clone()
+        try:
+            pdu_model.parse(message)
+            return pdu_model
+        except RecordFluxError as e:
+            return str(e)
 
-    message_name = str(identifier.name)
-    package_name = str(identifier.parent)
-    try:
-        pdu_message = PyRFLX.from_specs([str(path_spec)], skip_model_verification=no_verification)[
-            package_name
-        ][message_name]
-    except KeyError as e:
-        raise ValidationError(
-            f'message "{message_name}" could not be found in package "{package_name}"'
-        ) from e
-    except FileNotFoundError as e:
-        raise ValidationError(f"specification {e}") from e
-
-    classified_incorrectly = 0
-    with JsonOutputWriter(full_output_path) as json_output_writer:
-        validator = Validator(pdu_message)
-        for path, path_generator in path_generators.items():
-            for message_file_path in path_generator:
-                msg = get_message_to_validate(message_file_path, path == path_valid)
-                validation_result = validator.validate_message(msg)
-                json_output_writer.write(validation_result)
-                if validation_result.classification in [
-                    Classification.FP,
-                    Classification.FN,
-                ]:
-                    print(f"{message_file_path}: FAILED")
-                    print(f"{validation_result.get_abbreviated_output()}\n")
-                    if abort_on_error:
-                        raise ValidationError(
-                            f"{message_file_path} classified as "
-                            f"{validation_result.classification.value}"
-                        )
-                    classified_incorrectly += 1
-                else:
-                    print(f"{message_file_path}: PASSED")
-
-    if classified_incorrectly != 0:
-        raise ValidationError(f"{classified_incorrectly} messages were classified incorrectly")
-
-
-class JsonOutputWriter:
+    def validate(original_message: bytes, original_message_is_valid: bool, msg_path: Path) -> bool:
+        parser_result = rflx_parse_message_from_bytes(original_message)
```

Comment:
I merged the `rflx_parse_message_from_bytes` function, the `validate` function and the `get_message_from_path` function into one (in 60bb6d8 ) and created a class to encapsulate all data that results from one validation run. I know this class is probably "shallow" but since all this data is needed for the output, i think it is a better solution than having many arguments (in varying constellations) to the output functions. It might also be beneficial in the future for the coverage functionality since the class could be extended by an attribute describing the reached coverage of one message.